docs/frontend-system/architecture/04-plugins.md

```diff
@@ -10,84 +10,151 @@ description: Frontend plugins
 
 ## Introduction
 
-<!--
+In addition to the existing [`plugins`](../../plugins/index.md) documentation let's take a look at what changes for plugins in the new frontend system. If you already created a plugin yourself you will recognise a lot of similarity in the new architecture with the existing one.
 
-Introduce frontend plugins and how each of them is shipped as a separate package. Highlight that each package should only export a single plugin instance.
+Backstage is a single-page application composed of a set of plugins. Each of this plugins should solve exactly one responsibility & follow our suggested [plugin package structure & naming](../../architecture-decisions/adr011-plugin-package-structure.md).
 
--->
+For frontend plugins each plugin should only export a single plugin instance. The new frontend system can detect plugins, if they are exported as `default`, from the plugin package. This is also possible during runtime allowing you to add plugins to your Backstage instance without requiring a restart.
```

Comment:
```suggestion
Frontend plugins should only export a single plugin instance as its `default` export. This simplifies installation of the plugin, as the framework is able to discover default exports automatically.
```

Couple of tweaks, although it might need a bit more work since it got a bit short, maybe merge without the previous paragraph etc.

This is all related to frontend plugins, so no need to specify it as explicitly imo. Similarly, this is all also talking about the new frontend system. On that note it's best to keep these docs as standalone as possible so that we don't need to rewrite them again in the future. So rather than explaining things in terms of the existing system, try to go for a standalone explanation. We do still have the migration sections where we can explain things in terms of the old system.

Lastly I dropped the last sentence, that's referring to dynamic plugin installation right? We don't support that yet so best not to imply that we do. Although we could bring it back as a future promise though, although we'll definitely forget to update that xP